index.d.ts

```diff
@@ -114,28 +137,56 @@ export interface ResponseFailure {
   };
 }
 
-export interface Responses {
-  sent:   ResponseSent[];
-  failed: ResponseFailure[];
+export interface BroadcastResponseFailure extends Omit<ResponseFailure, "device"> {
+  bundleId: string;
+}
+
+export interface LoggerResponseFailure extends Partial<ResponseFailure>, Partial<BroadcastResponseFailure> {}
+
+export interface Responses<R,F> {
```

Comment:
I used generics here to:
1. Keep the original responses and not break code
2. Write less code for each type

I attempted to write this in a way to still work out-of-the-box with no breaking, but I'm not a TS expert, so feel free to chime in if you see an issue.